lib/rules/no-constant-condition.js

```diff
@@ -19,10 +19,15 @@ module.exports = function(context) {
     /**
      * Checks if a node has a constant truthiness value.
      * @param {ASTNode} node The AST node to check.
+     * @param {boolean} inBooleanPosition `false` if checking branch of a condition.
+     *  `true` by default.
      * @returns {Bool} true when node's truthiness is constant
      * @private
      */
-    function isConstant(node) {
+    function isConstant(node, inBooleanPosition) {
+        if (typeof inBooleanPosition === "undefined") {
```

Comment:
Sure, I will update passing `true` as you mentioned. Can you please explain the reason why `null` is preferred over `undefined`